src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,67 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileContent = readFileContent(fromFileName);
+        String report = createReport(fileContent);
+        writeToFile(toFileName, report);
+    }
+
+    public static String readFileContent(String filePath) {
+        StringBuilder fileContent = new StringBuilder();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                fileContent.append(line).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file " + filePath, e);
+        }
+
+        return fileContent.toString();
+    }
+
+    public static String createReport(String fileContent) {
+        int totalSupply = 0;
+        int totalBuy = 0;
 
+        String[] lines = fileContent.split(System.lineSeparator());
+        StringBuilder result = new StringBuilder();
+
+        for (String line : lines) {
+            String[] parts = line.split(",");
+            if (parts.length == 2) {
+                String action = parts[0].trim();
+                int value = Integer.parseInt(parts[1].trim());
```

Comment:
```suggestion
            String[] parts = line.split(DELIMITER);
            if (parts.length == CORRECT_PARTS_SIZE) {
                String action = parts[ACTION_INDEX].trim();
                int value = Integer.parseInt(parts[VALUE_INDEX].trim());
```

create constants for these values